modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -3645,12 +3650,81 @@ GDScriptParser::DataType GDScriptAnalyzer::make_global_class_meta_type(const Str
 	}
 }
 
+Ref<GDScriptParserRef> GDScriptAnalyzer::ensure_cached_parser_for_class(const GDScriptParser::ClassNode *p_class, const GDScriptParser::ClassNode *p_from_class, const String &p_context, const GDScriptParser::Node *p_source) {
+	if (p_class == nullptr) {
+		return nullptr;
+	}
+
+	if (parser->has_class(p_class)) {
+		return nullptr;
+	}
+
+	{
+		auto E = external_class_parser_cache.find(p_class);
```

Comment:
vnen made me notice this - we currently forbid using `auto` and prefer the type to be explicit, unless it's a special edge case where `auto` solves a language constraint.

https://docs.godotengine.org/en/stable/contributing/development/cpp_usage_guidelines.html#auto-keyword